quicklens/src/main/scala-3/com/softwaremill/quicklens/QuicklensMacros.scala

```diff
@@ -430,7 +408,7 @@ object QuicklensMacros {
         objTerm
 
       case (_: (PathSymbol.Field | PathSymbol.Extension), _) :: _ =>
-        val (fs, funs) = pathSymbols.span(s => s._1.isInstanceOf[PathSymbol.Field] | s._1.isInstanceOf[PathSymbol.Extension])
+        val (fs, funs) = pathSymbols.partition((ps, _) => ps.isInstanceOf[PathSymbol.Field] || ps.isInstanceOf[PathSymbol.Extension])
```

Comment:
`span` vs `partition` is quite different. Why is `partition` correct now, or was `span` always a mistake? I do not understand this part of the code much, therefore I am unable to judge.

Reply:
Oooh, I might've changed things too fast. Thought that this was a bigger change. I'll revert this back. `partition` and `span` can both work here, but it depends on the guarantees we want to have, mostly for side effecting code.